app/src/main/java/org/simple/clinic/registration/phone/RegistrationPhoneEffectHandler.kt

```diff
@@ -20,6 +27,29 @@ class RegistrationPhoneEffectHandler @AssistedInject constructor(
     return RxMobius
         .subtypeEffectHandler<RegistrationPhoneEffect, RegistrationPhoneEvent>()
         .addConsumer(PrefillFields::class.java, { uiActions.preFillUserDetails(it.entry) }, schedulers.ui())
+        .addTransformer(LoadCurrentRegistrationEntry::class.java, loadCurrentRegistrationEntry())
         .build()
   }
+
+  private fun loadCurrentRegistrationEntry(): ObservableTransformer<LoadCurrentRegistrationEntry, RegistrationPhoneEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .switchMapSingle { currentOngoingRegistrationEntry() }
+          .map(::CurrentRegistrationEntryLoaded)
+    }
+  }
+
+  private fun currentOngoingRegistrationEntry(): Single<Optional<OngoingRegistrationEntry>> {
+    return userSession
+        .isOngoingRegistrationEntryPresent()
+        .flatMap { isRegistrationEntryPresent ->
+          // TODO (vs) 04/06/20: This is nasty, make it a synchronous call
+          if (isRegistrationEntryPresent)
+            userSession
+                .ongoingRegistrationEntry()
+                .map { Just(it) }
```

Comment:
You're right, I left this as is since I was copying over code from the controller. However, in a later part of this migration, I have already made this change so this will be reflected in another PR.